code/clue.js

```diff
@@ -41,19 +189,73 @@ const randomSelector = array => {
 // With a killer, a weapon and a room.
 // The values will be set later.
 
+let mystery = {
+  killer:'',
+  weapon:'',
+  room:''
+}
+
+
+
+
 // This function will be invoked when you click on the killer card.
+
 const pickKiller = () => {
   // This will randomly select a killer from the suspects. And add that to the mystery object.
   mystery.killer = randomSelector(suspects)
+  
+  // Shuffle weapon function
+  shuffleFavouriteWeapon = () => {
+    mystery.killer.favouriteWeapon = randomSelector(weapons); 
+  };
+    shuffleFavouriteWeapon();
+    console.log(mystery.killer.favouriteWeapon.name)
 
   // This will change the background color of the card to the one connected to the chosen killer and show the full name of the killer. Feel free to add more things to show about the killer.
   document.getElementById('killerCard').style.background = mystery.killer.color
+  document.getElementById('killerImage').src = mystery.killer.image
   document.getElementById(
     'killerName'
   ).innerHTML = `${mystery.killer.firstName} ${mystery.killer.lastName}`
+  document.getElementById('killerFavoriteWeapon').innerHTML = `${mystery.killer.favouriteWeapon.name}`
 }
+document.getElementById('killerCard').addEventListener('click', pickKiller)
```

Comment:
Another way to write this would be `document.getElementById('killerCard').onclick = pickKiller`

Reply:
Yes, but it's alternative, you can choose whatever way you want